single_kernel_mongo/managers/ldap.py

```diff
@@ -191,4 +194,36 @@ def get_status(self) -> StatusBase | None:
                 logger.info("Waiting for LDAP data.")
                 return BlockedStatus("Missing LDAP data from Glauth.")
             case _:
-                return get_ldap_connection_status(self.state.ldap, self.state.paths)
+                return self.get_ldap_connection_status()
+
+    def get_ldap_connection_status(self) -> StatusBase:
+        """Checks if the LDAP connection is working or not.
+
+        Helpful to prevent restarts that would fail.
+        """
+        bind_dn = self.state.ldap.bind_user
+        bind_password = self.state.ldap.bind_password
+        base_dn = self.state.ldap.base_dn
+
+        if not base_dn:
+            return BlockedStatus("Missing base DN.")
+
+        if not self.state.ldap.ldaps_urls:
+            return BlockedStatus("Missing LDAPS URLs.")
+
+        try:
+            for ldap_uri in self.state.ldap.ldaps_urls:
+                tls = LDAPTls(
+                    validate=ssl.CERT_REQUIRED,
+                    version=ssl.PROTOCOL_TLSv1_2,
+                    ca_certs_file=f"{self.state.paths.ldap_certificates_file}",
+                )
+                server = LDAPServer(host=ldap_uri, use_ssl=True, tls=tls)
+                conn = LDAPConnection(server, user=bind_dn, password=bind_password)
+                conn.bind()  # We consider sufficient to be able to bind.
+                conn.unbind()
+        except Exception as err:
```

Comment:
perhaps we can check for a specific error?

Reply:
Sure, I'll improve that.